File: April/Dhruvabhat24/C++/April 01/Solution.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int n = s.size(), length = 0;
        for(int i=n-1;i>=0;i--){
            if(length>0 && s[i] == ' ')break;
            if(s[i] != ' ')length++;
        }
        return length;
    }
};

File: April/Dhruvabhat24/Java/April 01/Solution.java
class Solution {
  public int lengthOfLastWord(String s) {
    int i = s.length() - 1;

    // Move pointer to the left while encountering consecutive spaces
    while (i >= 0 && s.charAt(i) == ' ')
      --i;

    // Set lastIndex to the current position of i
    final int lastIndex = i;

    // Move pointer to the left until the start of the last word
    while (i >= 0 && s.charAt(i) != ' ')
      --i;

    // Calculate the length of the last word
    return lastIndex - i;
  }
}
